src/routes/cache-hooks/cache-hooks.controller.ts

```diff
@@ -14,26 +14,35 @@ import { Event } from '@/routes/cache-hooks/entities/event.entity';
 import { PreExecutionLogGuard } from '@/routes/cache-hooks/guards/pre-execution.guard';
 import { WebHookSchema } from '@/routes/cache-hooks/entities/schemas/web-hook.schema';
 import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { IConfigurationService } from '@/config/configuration.service.interface';
 
 @Controller({
   path: '',
   version: '1',
 })
 @ApiExcludeController()
 export class CacheHooksController {
+  private readonly isEventHooksEndpointEnabled: boolean;
   constructor(
     private readonly service: CacheHooksService,
     @Inject(LoggingService) private readonly loggingService: ILoggingService,
-  ) {}
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+  ) {
+    this.isEventHooksEndpointEnabled =
+      this.configurationService.getOrThrow<boolean>('features.eventsQueue');
+  }
 
   @UseGuards(PreExecutionLogGuard, BasicAuthGuard)
   @Post('/hooks/events')
   @HttpCode(202)
   async postEvent(
     @Body(new ValidationPipe(WebHookSchema)) event: Event,
   ): Promise<void> {
-    this.service.onEvent(event).catch((error) => {
-      this.loggingService.error(error);
-    });
+    if (!this.isEventHooksEndpointEnabled) {
+      this.service.onEvent(event).catch((error) => {
+        this.loggingService.error(error);
+      });
+    }
```

Comment:
Here I opted for just disabling the reception of events via HTTP _silently_, instead of unmounting the controller, to avoid disrupting the Events Service, which theoretically would still send HTTP requests when the queue consumption is activated.